### PR TITLE
Bugfix/clean storage meta data table

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
once upon a time, storage micro-service was not removing the entries in the file_meta_data table correctly after deleting projects.

This PR aims to bring a special maintenant API entrypoint in storage that triggers a synchronisation of the table with the actual S3 backend by checking:
- each entry of the file_meta_data table is compared against S3
- if an entry is not visible in S3, it is then added to the "to be removed" list
- if the call is not in dry-run mode, each entry will be removed from the table
- the list of files to removed is returned

The endpoint is available form the webserver for TESTER only.

## Related issue/s
solves [#411](https://github.com/ITISFoundation/osparc-issues/issues/411)

## How to test

1. Upload some files to osparc through f.ex. file-picker ```http://127.0.0.1:9081```
2. Log into S3 ```http://127.0.0.1:9001```
3. remove the file(s) through minio interface
4. go back to osparc, edit the url by adding ```/dev/doc``` (this will automatically log as yourself in the Swagger UI)
5. Go to POST ```/v0/storage/locations/{id}:sync```
6. Set 0 as location id
7. Run, you should receive a list of files with the ones to be removed
8. Run with query parameter ```dry_run``` as false, this will effectively delete the entries in the DB

### CODE REVIEW

File: services/storage/src/simcore_service_storage/dsm.py [hunk 1]
```diff
@@ -1009,3 +1009,42 @@ async def create_soft_link(
...
+                        await s3_client.get_object(
+                            Bucket=self.simcore_bucket_name,
+                            Key=s3_key,
+                        )
+                    except s3_client.exceptions.NoSuchKey:
+                        # this file does not exist
+                        sync_results["removed"].append(s3_key)
+
+                if not dry_run:
+                    await conn.execute(
```
Comment (pcrespov) [inline services/storage/src/simcore_service_storage/dsm.py:1044]: fire & forget?

File: services/web/server/src/simcore_service_webserver/security_roles.py [hunk 2]
```diff
@@ -78,6 +78,7 @@
     UserRole.TESTER: {
         "can": [
             "diagnostics.read",
+            "storage.files.sync",
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/security_roles.py:81]: For later: shouldn't we create a higher user role like admin for these operations? It is an operation assigned to an framework administrator more than a tester

**Review by pcrespov — APPROVED** [review]
:+1:

File: services/web/server/src/simcore_service_webserver/storage_handlers.py [hunk 3]
```diff
@@ -89,69 +89,76 @@ def extract_link(data: Optional[Dict]) -> str:
...
     return payload
 
 
+@login_required
+@permission_required("storage.files.sync")
+async def synchronise_meta_data_table(request: web.Request):
+    payload = await _request_storage(request, "POST")
+    return payload
+
+
```
Comment (GitHK) [inline services/web/server/src/simcore_service_webserver/storage_handlers.py:161]: MINIOR: only if you have to restart the CI, put this on the top of the file

**Review by GitHK — APPROVED** [review]
👍

File: services/web/server/src/simcore_service_webserver/security_roles.py (hunk 2 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/security_roles.py:81]: yes. that would make sense.

File: services/web/server/src/simcore_service_webserver/storage_handlers.py (hunk 3 above)
Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/storage_handlers.py:161]: not sure I understand why?

Comment (GitHK) [reply services/web/server/src/simcore_service_webserver/storage_handlers.py:161]: never mind, I've read it wrong

**Review by odeimaiz — APPROVED** [review]
👌

**Pull request merged**